apps/embed-web/src/signOperationModalHooks.tsx

```diff
@@ -0,0 +1,49 @@
+import { type PartialTezosOperation } from "@airgap/beacon-sdk";
+import { Center, Modal, ModalCloseButton, ModalContent, useDisclosure } from "@chakra-ui/react";
+import { useState } from "react";
+
+import { OperationModalContent } from "./SignOperationModalContent";
+import { sendOperationErrorResponse } from "./utils";
+
+export const useSignOperationModal = () => {
+  const { isOpen, onOpen, onClose } = useDisclosure();
+  const [operations, setOperations] = useState<PartialTezosOperation[]>([]);
+
+  const onModalCLose = () => {
+    sendOperationErrorResponse("User closed the modal");
+    onClose();
+  };
+
+  return {
+    modalElement: (
+      <Center>
+        <Modal
+          autoFocus={false}
+          closeOnOverlayClick={false}
+          isCentered
+          isOpen={isOpen}
+          onClose={onClose}
+        >
+          <ModalContent>
+            <ModalCloseButton onClick={onModalCLose} />
+            <OperationModalContent closeModal={onClose} operations={operations} />
+          </ModalContent>
+        </Modal>
+      </Center>
+    ),
+    onOpen: (operations: PartialTezosOperation[]) => {
+      // TODO: if (isValid(operations)) {
+      // TODO: get fee
+      setOperations(operations);
+      onOpen();
+    },
+  };
+};
+
+// TODO: updated to validate all types of operations
```

Comment:
please remove this block because [this](https://www.opsera.io/blog/what-is-code-smell#:~:text=Code%20comments%20are%20a%20code,place%20is%20a%20code%20smell.)

Reply:
Done, thank you!